ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/observer/DelegatedCall.kt

```diff
@@ -40,6 +40,25 @@ public fun HttpClientCall.wrap(content: ByteReadChannel, headers: Headers): Http
     return DelegatedCall(client, content, this, headers)
 }
 
+/**
+ * Wrap existing [HttpClientCall] with new [headers] and content produced by the given [block].
+ * The [block] will be called each time the response content is requested.
+ *
+ * ```
+ * // Create new response with modified headers and content
+ * val headers = headersOf("Custom-Header", "Custom-Value")
+ * val newCall = originalCall.wrap(headers) {
+ *     ByteReadChannel("Modified content")
+ * }
```

Comment:
@coderabbitai I've failed to come up with a useful example here. Do you have any suggestions?

Reply:
Thank you, fixed the example